realm-annotations-processor/src/main/java/io/realm/processor/RealmVersionChecker.java

```diff
@@ -33,10 +35,17 @@
 
     private ProcessingEnvironment processingEnvironment;
 
-    public RealmVersionChecker(ProcessingEnvironment processingEnvironment) {
+    private RealmVersionChecker(ProcessingEnvironment processingEnvironment) {
         this.processingEnvironment = processingEnvironment;
     }
 
+    public static RealmVersionChecker getInstance(ProcessingEnvironment processingEnvironment) {
```

Comment:
Using a singleton pattern this is usually synchronized, but since it is only running as part of the annotation processor I guess it is OK.